components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java

```diff
@@ -2411,6 +2417,64 @@ public Endpoint getEndpointByName(String name) throws APIMgtDAOException {
         return endpointList;
     }
 
+    /**
+     * This returns the json string containing the role permissions for a given API
+     * @param connection - DB connection
+     * @param apiId - apiId of the API
+     * @return permission string
+     * @throws SQLException - if error occurred while getting permissionMap of API from DB
+     */
+    private StringBuilder getPermissionsStringForApi(Connection connection, String apiId) throws SQLException {
+        JSONArray permissionArray = new JSONArray();
+        Map<String, Integer> permissionMap = getPermissionMapForApi(connection, apiId);
+        for (Map.Entry<String, Integer> entry : permissionMap.entrySet()) {
+            JSONObject jsonObject = new JSONObject();
+            jsonObject.put(APIMgtConstants.Permission.GROUP_ID, entry.getKey());
+            ArrayList<String> array = new ArrayList<String>();
+            Integer permissionValue = entry.getValue();
+            if (permissionValue == APIMgtConstants.Permission.READ_PERMISSION) {
+                array.add(APIMgtConstants.Permission.READ);
+            } else if (permissionValue == (APIMgtConstants.Permission.READ_PERMISSION
+                    + APIMgtConstants.Permission.UPDATE_PERMISSION)) {
+                array.add(APIMgtConstants.Permission.READ);
+                array.add(APIMgtConstants.Permission.UPDATE);
+            } else if (permissionValue == (APIMgtConstants.Permission.READ_PERMISSION
+                    + APIMgtConstants.Permission.DELETE_PERMISSION)) {
+                array.add(APIMgtConstants.Permission.READ);
+                array.add(APIMgtConstants.Permission.DELETE);
+            } else if (permissionValue == (APIMgtConstants.Permission.READ_PERMISSION
+                    + APIMgtConstants.Permission.UPDATE_PERMISSION + APIMgtConstants.Permission.DELETE_PERMISSION)) {
+                array.add(APIMgtConstants.Permission.READ);
+                array.add(APIMgtConstants.Permission.UPDATE);
+                array.add(APIMgtConstants.Permission.DELETE);
+            }
+            jsonObject.put(APIMgtConstants.Permission.PERMISSION, array);
+            permissionArray.add(jsonObject);
+        }
+        return new StringBuilder(permissionArray.toString());
+    }
+
+    /**
+     * This constructs and returns the API permissions map from the DB
```

Comment:
Let's add a new line to make this consistant with other java doc comments. Make this change in other places as well.